app/views/public_recipes/index.html.erb

```diff
@@ -0,0 +1,23 @@
+<main>
+    <h1 class="title">Public Recipes</h1>
+    <% if @recipes.blank? %>
+    <p>There are not any public recipes at the moment</p>
+    <% else %>
+    <ul class="public_recipes">
+    <% @recipes.each do |recipe| %>
+      <%= link_to user_recipe_path(recipe.user.id, recipe), class: "link" do %>
+        <li class="public_list_item">
+            <div class="item" id="item">
+              <h2><%= recipe.name %></h2>
+              <h4>By: <%= recipe.user.name %></h4>
+            </div>
+            <div class="item">
+                <p>total food items: <strong><%= recipe.user.foods.count %></strong></p>
+                <p>total price: <strong>$<%= recipe.user.foods.sum { |food| food.price * food.quantity } %></strong></p>
+            </div>
+        </li>
+      <%end%>
+    <% end %>
+    </ul>
+    <% end %>
+</main>
```

Comment:
When I am not logged in, I am still able to see the public recipes, and when I click on the recipe, it leads to this error.
![image](https://github.com/ernestmusong/Recipe-app/assets/33268927/3d54f190-4e63-4278-9a66-b26d4f86e766)

Reply:
Thank you @muneebulrehman for pointing out this error. However, the error has been fixed but I have deliberately allowed the public recipes to be accessible by all users both logged-in and non logged-in users. In this case, non logged-in users have read-only capabilities.

Reply:
well resolved! ✔️